tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -37,93 +43,107 @@ public static void OnInstrumentPublished(Instrument instrument, MeterListener li
         }
 
         bool shouldEnable;
-        var enabledMeterNames = Tracer.Instance.Settings.OpenTelemetryMeterNames;
+        var enabledMeterNames = _settings.OpenTelemetryMeterNames;
```

Comment:
nit: It _might_ be more performant to make `TracerSettings.OpenTelemetryMeterNames` a `HashSet<string>` to improve the lookup time. This is a potential followup item